staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -96,8 +96,11 @@ func WithAuthentication(handler http.Handler, auth authenticator.Request, failed
 			return
 		}
 
-		// TODO(mikedanese): verify the response audience matches one of apiAuds if
-		// non-empty
+		if len(apiAuds) > 0 && resp != nil && len(resp.Audiences) > 0 && len(authenticator.Audiences(apiAuds).Intersect(resp.Audiences)) == 0 {
+			klog.Errorf("Unable to match the audience: %v", resp.Audiences)
```

Comment:
this is an error case for the authenticator, consider logging apiAuds as well to ease resolution